src/rafimage.cpp

```diff
@@ -364,17 +364,25 @@ namespace Exiv2 {
         // sanity check.  Does tiff lie inside the file?
         enforce(Safe::add(tiffOffset, tiffLength) <= io_->size(), kerCorruptedMetadata);
 
-        DataBuf  tiff(tiffLength);
         if (io_->seek(tiffOffset, BasicIo::beg) != 0) throw Error(kerFailedToReadImageData);
-        io_->read(tiff.pData_, tiff.size_);
 
-        if (!io_->error() && !io_->eof())
+        // check if this really is a tiff (some older models just embed a raw bitstream)
```

Comment:
Just wanted to make it a bit more clear that the behavior we want for the raw bitstream case is a "no-op"

```suggestion
        // Check if this really is a tiff and then call the tiff parser.
        // Check is needed because some older models just embed a raw bitstream.
        // For those files we skip the parsing step. 
```